### PR TITLE
Basic Testing Infrastructure

### PR DESCRIPTION
**Pull request by tsalh**

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+"""
+This is a test class to assess the functionality of the data structures 
+defined in util.py. The data structures to be tested are the Queue, Stack,
+and the Priority Queue
+"""
+class UtilTest(unittest.TestCase):
+
+    def test_queue(self):
+
```
Comment (cfpryor) [inline tests/test_util.py]: Extra Space

File: tests/test_util.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        for val in random_val_list:
+            queue.push(val)
+        self.assertFalse(queue.isEmpty())
+
+        # Test Queue for FIFO functionality
+        for val in random_val_list:
+            self.assertEqual(val, queue.pop())
+
+    def test_stack(self):
+
```
Comment (cfpryor) [inline tests/test_util.py]: Extra space

File: tests/test_util.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from pacai.util import util
+
+"""
+This is a test class to assess the functionality of the data structures 
+defined in util.py. The data structures to be tested are the Queue, Stack,
+and the Priority Queue
+"""
+class UtilTest(unittest.TestCase):
+
```
Comment (cfpryor) [inline tests/test_util.py]: Extra Space

File: tests/test_util.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        for val in random_val_list:
+            stack.push(val)
+        self.assertFalse(stack.isEmpty())
+
+        # Test Stack for LIFO functionality
+        for val in reversed(random_val_list):
+            self.assertEqual(val, stack.pop())
+
+    def test_priority_queue(self):
+
```
Comment (cfpryor) [inline tests/test_util.py]: Extra Space

File: tests/test_util.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+and the Priority Queue
+"""
+class UtilTest(unittest.TestCase):
+
+    def test_queue(self):
+
+        queue = util.Queue()
+        self.assertTrue(queue.isEmpty())
+
+        random_val_list = [hash(x) for x in range(1, 10)]
```
Comment (cfpryor) [inline tests/test_util.py]: Why are you doing hash? This is an int, so hash will just return the int value.

File: tests/test_util.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        # position in the list
+        random_val_list = [(hash(x), x) for x in range(1, 10)]
+        for val, pri in random_val_list:
+            priority_queue.push(val, pri)
+        self.assertFalse(priority_queue.isEmpty())
+
+        # Test Min Queue for priority functionality
+        for val, pri in random_val_list:
+            self.assertEqual(val, priority_queue.pop())
+
```
Comment (cfpryor) [inline tests/test_util.py]: Extra Space

File: tests/test_util.py (hunk 5 above)
Comment (tsalh) [reply tests/test_util.py]: I just wanted to fill the list with random values, no significant reason as to why I used hash().

Comment (eriq-augustine) [reply tests/test_util.py]: hash() on an int just returns the int's value.

Comment (eriq-augustine) [reply tests/test_util.py]: Also, be VERY careful using random values in tests. Avoid it whenever you can.
For example, you also use hash() in the priority queue test below. If hash() did return an arbitrary value (say an address), then that test would randomly fail (we call it a flaky test).

Comment (tsalh) [reply tests/test_util.py]: Understood

File: tests/test_bin.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import unittest
+
+from pacai.bin import pacman
```
Comment (cfpryor) [inline tests/test_bin.py]: spacing and switch order

Comment (eriq-augustine) [reply tests/test_bin.py]: I believe that Connor means to make sure that imports in the same group are sorted alphabetically.
(Not to swap lines 1 and 3.)

Comment (tsalh) [reply tests/test_bin.py]: I'm confused here, so is all I have to do is remove the line in between the imports? If I am not swapping lines 1 and 3.

Comment (eriq-augustine) [reply tests/test_bin.py]: Imports should be in alphabetical order.
So remove the extra line and swap the order of the `pacai.bin` imports.

**Pull request merged**